site/content/contribute/server/style-guide.md

```diff
@@ -0,0 +1,98 @@
+
+Avoid creating one-off goroutines without knowing when/how they exit. They cause problems that are hard to debug, and can often cause performance degradation rather than an improvement. Have a look at:
+- https://github.com/golang/go/wiki/CodeReviewComments#goroutine-lifetimes
+- https://github.com/golang/go/wiki/CodeReviewComments#synchronous-functions
+
+### Pointers to slices (exempt from model)
+
+Avoid using pointers to slices. Slices are already reference types which point to an underlying array. If you want a function to modify a slice, then return that slice from the function, rather than passing a pointer.
+
+### [CamelCase variables/constants](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps) (exempt from model)
+
+We prefer to use names like WebsocketEventPostEdited, not WEBSOCKET_EVENT_POST_EDITED.
+
+### Empty string check
+
+Use == "" to check empty strings, not len(s) == 0. This will be enforced with a linter soon.
+
+### [Reduce indentation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
+
+If there are multiple return statements in an if-else statement, remove the else block and outdent it.
+
+This is a sample code from `mlog/human/parser.go`:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+} else {
+	d, ok := token.(json.Delim)
+	if !ok || d != '{' {
+		return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+	}
+}
+```
+
+This can be simplified to:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+}
+d, ok := token.(json.Delim)
+if !ok || d != '{' {
+	return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+}
+```
+
+### Avoid creating more ToJSON methods
+
+We should avoid creating ToJSON methods for model structs. And just use `json.Marshal` at the call site. This has two main benefits:
+- We’ve had to fix multiple bugs due to suppressing the JSON error which happens with ToJSON methods.
+- A lot of times, we pass the output to something (like a network call) which accepts a byte slice, leading to a double conversion from byte-slice to string to a byte-slice again.
+
+### [Initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) (exempt from model)
+
+We prefer to use userID rather than userId. Same for abbreviations; HTTP is preferred over Http or http. This will be enforced with the linter soon.
+
+### [Interfaces](https://github.com/golang/go/wiki/CodeReviewComments#interfaces)
+
+- Return structs, accept interfaces.
+- Interface names should end with “-er”. This is not a strict rule. Just a guideline which indicates the fact that interface functionalities are designed around the concept of “doing” something.
+- Try not to define interfaces on the implementer side of an API "for mocking"; instead, design the API so that it can be tested using the public API of the real implementation.
```

Comment:
Not blocking for this review, but I think particularly the latter two points in this list would benefit from some sample code.

Reply:
Added in this review itself.